clang/lib/AST/DeclTemplate.cpp

```diff
@@ -1598,6 +1599,16 @@ BuiltinTemplateDecl::BuiltinTemplateDecl(const ASTContext &C, DeclContext *DC,
                    createBuiltinTemplateParameterList(C, DC, BTK)),
       BTK(BTK) {}
 
+bool BuiltinTemplateDecl::isPackProducingBuiltinTemplate() const {
+  return getBuiltinTemplateKind() == clang::BTK__builtin_dedup_pack;
+}
+
+bool clang::isPackProducingBuiltinTemplateName(TemplateName N) {
+  auto *T = dyn_cast_or_null<BuiltinTemplateDecl>(
+      N.getTemplateDeclAndDefaultArgs().first);
```

Comment:
Small nit, slightly cleaner:
```suggestion
  auto *T = dyn_cast_or_null<BuiltinTemplateDecl>(
      N.getAsTemplateDecl(/*IgnoreDeduced=*/true));
```

Reply:
FYI the performance of this is also better, since we won't build the defaulted argument list.

Reply:
Thanks for spotting this! Done.